### ISSUE TITLE
Inline editor: Issues when the root element is a button

### ISSUE THREAD

**Issue by nicole-ashley** [tinymce/tinymce#5098]
**What is the current behavior? Describe the bug**
When tinymce inline is initialised on a button (the intent being to allow the button label to be edited) there are a few issues:

1. The button's content is cleared when the editor is initialised
1. Clicking on the editable area activates the button action
1. Pressing spacebar while editing activates the button action and in some cases prevents the space from being inserted

Issue 1 seems to be resolved on Chrome by making a child span the root element, but this does not resolve issues 2 and 3 and completely breaks editing on Firefox. See http://fiddle.tinymce.com/LNgaab

**What is the expected behavior?**

1. The button's content should not be cleared when the editor is initialised
1. Clicking on the editable area should not activate the button action
1. Pressing spacebar while editing should not activate the button action, and the space should be inserted

While issues 2 and 3 could arguably be the responsibility of the implementing developer, issue 1 is definitely a TinyMCE bug. Fixes for issues 2 and 3 would be nice as well - if and when a button is editable I would never expect it to be actually used as a button.

**Which versions of TinyMCE, and which browser / OS are affected by this issue? Did this work in previous versions of TinyMCE?**

TinyMCE 5 Cloud (stable)
Tested on latest versions of Chrome, Firefox, and Edge on Windows 10.

**Comment by IamFonky** [comment]
``` HTML
<button type="button" class="mce" value="This seem to work"/>
```
Does this line helps? It worked well for me on firefox

**Comment by nikrolls** [comment]
Thanks @IamFonky. That would potentially work, but then that value will be submitted with the form. Buttons are meant to have content, and I'd like to edit that content rather than the value.

**Comment by lnewson** [comment]
I can't comment on points 2 & 3 at this stage, as that'd need more investigation, but point 1) was actually fixed in 5.0.10 (see https://github.com/tinymce/tinymce/issues/4934). Here's a fiddle that uses cloud 5-dev instead of 5-stable, in which you can see the text content is retained: http://fiddle.tinymce.com/LNgaab/1

**Comment by nikrolls** [comment]
Ah! Thank you @lnewson, I was clearly searching too narrowly.

Points 2 & 3 I have mitigated in my own code for now, with a preventDefault and an insert. Though it may be worth a discussion as to whether this should be handled by default (I'm sure other elements could benefit from it).
